File: app/src/main/java/c/m/storyapp/register/domain/repository/RegisterRepository.kt
package c.m.storyapp.register.domain.repository

import c.m.storyapp.common.util.Resource
import c.m.storyapp.register.domain.model.RegisterResponse
import kotlinx.coroutines.flow.Flow

interface RegisterRepository {
    suspend fun postRegister(name: String, email: String, password: String): Flow<Resource<RegisterResponse>>
}

File: lib/register/domain/repository/register_repository.dart
import 'package:story_app/common/util/resource.dart';
import 'package:story_app/register/domain/model/register_response.dart';

abstract class RegisterRepository {
  Future<Resource<RegisterResponse>> postRegister(
    String name,
    String email,
    String password,
  );
}
